devices/bme280/bme280.go

```diff
@@ -31,113 +34,209 @@ import (
+// consumption may increase the temperature reading.
 const (
-	No   Oversampling = 0
+	Off  Oversampling = 0
 	O1x  Oversampling = 1
 	O2x  Oversampling = 2
 	O4x  Oversampling = 3
 	O8x  Oversampling = 4
 	O16x Oversampling = 5
 )
 
-// Standby is the time the BME280 waits idle between measurements. This reduces
-// power consumption when the host won't read the values as fast as the
-// measurements are done.
-type Standby uint8
+const oversamplingName = "Off1x2x4x8x16x"
 
-// Possible standby values, these determines the refresh rate.
-const (
-	S500us Standby = 0
-	S10ms  Standby = 6
-	S20ms  Standby = 7
-	S62ms  Standby = 1
-	S125ms Standby = 2
-	S250ms Standby = 3
-	S500ms Standby = 4
-	S1s    Standby = 5
-)
+var oversamplingIndex = [...]uint8{0, 3, 5, 7, 9, 11, 14}
+
+func (i Oversampling) String() string {
+	if i >= Oversampling(len(oversamplingIndex)-1) {
+		return fmt.Sprintf("Oversampling(%d)", i)
+	}
+	return oversamplingName[oversamplingIndex[i]:oversamplingIndex[i+1]]
+}
+
+func (o Oversampling) asValue() int {
+	switch o {
+	case O1x:
+		return 1
+	case O2x:
+		return 2
+	case O4x:
+		return 4
+	case O8x:
+		return 8
+	case O16x:
+		return 16
+	default:
+		return 0
+	}
+}
 
-// Filter specifies the internal IIR filter to get steady measurements without
-// using oversampling. This is mainly used to reduce power consumption.
+// Filter specifies the internal IIR filter to get steadier measurements.
+//
+// Oversampling will get better measurements than filtering but at a larger
+// power consumption cost, which may slightly affect temperature measurement.
 type Filter uint8
 
 // Possible filtering values.
+//
+// The higher the filter, the slower the value converges but the more stable
+// the measurement is.
 const (
-	FOff Filter = 0
-	F2   Filter = 1
-	F4   Filter = 2
-	F8   Filter = 3
-	F16  Filter = 4
+	NoFilter Filter = 0
+	F2       Filter = 1
+	F4       Filter = 2
+	F8       Filter = 3
+	F16      Filter = 4
 )
 
-// Dev is an handle to a bme280.
+// Dev is an handle to an initialized bme280.
 type Dev struct {
-	d     conn.Conn
-	isSPI bool
-	c     calibration
+	d         conn.Conn
+	isSPI     bool
+	opts      Opts
+	measDelay time.Duration
+	c         calibration
+
+	mu   sync.Mutex
+	stop chan struct{}
 }
 
 func (d *Dev) String() string {
 	return fmt.Sprintf("BME280{%s}", d.d)
 }
 
-// Sense returns measurements as °C, kPa and % of relative humidity.
+// Sense request a single one time measurement as °C, kPa and % of relative
+// humidity.
+//
+// The very first measurements may be of poor quality.
 func (d *Dev) Sense(env *devices.Environment) error {
-	// All registers must be read in a single pass, as noted at page 21, section
-	// 4.1.
-	// Pressure: 0xF7~0xF9
-	// Temperature: 0xFA~0xFC
-	// Humidity: 0xFD~0xFE
-	buf := [0xFF - 0xF7]byte{}
-	if err := d.readReg(0xF7, buf[:]); err != nil {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	if d.stop != nil {
+		return errors.New("bme280: already sensing continuously")
```

Comment:
My bad, the lock seems to be released when running continuously in the background. I'm now kind of confused about the use case of the mutex. I thought it was a way to protect the user from running multiple polls in parallel.